### PR TITLE
libafl_sugar

### PR DESCRIPTION
**Pull request by andreafioraldi**
It is a lib in which we put ready-to-use fuzzers based on libafl that the naive user can just use without knowing LibAFL under the hood

### CODE REVIEW

**Comment by andreafioraldi** [comment]
Example: https://github.com/AFLplusplus/LibAFL/blob/high_level/fuzzers/libfuzzer_stb_image_sugar/src/main.rs#L34

**Comment by domenukk** [comment]
I don't know if I would actually name them `..Sugar`, apart from that it looks great!
I'd probably rename InMemoryBytesCoverageSugar -> InMemoryBytesCoverageFuzzer (or something)

Any reason not to put this in the main lib?

File: fuzzers/baby_no_std/log [hunk 1]
```diff
@@ -0,0 +1,60 @@
+   Compiling stable_deref_trait v1.2.0
```
Comment (domenukk) [inline fuzzers/baby_no_std/log:1]: This file shouldn't be checked in

File: fuzzers/libfuzzer_stb_image_sugar/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Libfuzzer for stb_image
```
Comment (domenukk) [inline fuzzers/libfuzzer_stb_image_sugar/README.md:1]: Explain what's `sugar` about it

File: fuzzers/libfuzzer_stb_image_sugar/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+/// The actual fuzzer
+fn fuzz(input_dirs: &[PathBuf], output_dir: PathBuf, cores: &[usize], broker_port: u16) {
+    // Call LLVMFUzzerInitialize() if present.
+    let args: Vec<String> = env::args().collect();
+    if libfuzzer_initialize(&args) == -1 {
+        println!("Warning: LLVMFuzzerInitialize failed with -1")
+    }
+
+    InMemoryBytesCoverageSugar::builder()
```
Comment (domenukk) [inline fuzzers/libfuzzer_stb_image_sugar/src/main.rs:34]: looks great

**Comment by andreafioraldi** [comment]
>  I'd probably rename InMemoryBytesCoverageSugar -> InMemoryBytesCoverageFuzzer (or something)

Then it seems a derivation of the Fuzzer trait, that is wrong. Sugar sounds good to me because these wrapper are adding nothing, like syntax sugar.

> Any reason not to put this in the main lib?

Because we want to provide ready-to-use fuzzers so they need to import the instrumentation libs too (libafl_targets in this basic case)

**Comment by domenukk** [comment]
Name them Builder?

**Comment by domenukk** [comment]
Sugar is fine. Any plans to mege?

**Pull request merged**